src/testharness/TestHarness.cpp

```diff
@@ -83,12 +128,24 @@ bool TestHarness::runTestsForToolChain(std::string exeName, std::string tcName)
 
       // Iterate over each test in the package
       for (size_t i = 0; i < subPackage.size(); ++i) {
-        std::unique_ptr<TestFile>& test = subPackage[i];
+        TestPair& pair = subPackage[i];
+        std::unique_ptr<TestFile>& test = pair.first;
         if (test->getParseError() == ParseError::NoError) {
-        
-          TestResult result = runTest(test.get(), toolChain, cfg);
+
+          // Poll while we wait for the result
+          // TODO this could probably be replaced with some sort of interrupt,
+          //  (and probably should be), but better this than no threads
+          while (!pair.second.has_value())
+            std::this_thread::sleep_for(std::chrono::milliseconds(100));
+
+          TestResult result = pair.second.value();
+
+          // keep the result with the test for pretty printing
+          std::optional<TestResult> res_clone = std::make_optional(result.clone());
+          subPackage[i].second.swap(res_clone);
```

Comment:
These three lines don't do anything.

`pair` is `subPackage[i]`. `result` is a duplicate of `subPackage[i].second.value`. `res_clone` is a an optional of `subPackage[i].second.value`. Then you swap `subPackage[i].second` with that value, which results in no change.

Can you explain what the intent is here?